src/ORM/Association/HasMany.php

```diff
@@ -134,6 +181,67 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
     }
 
     /**
+     * Deletes/sets null the related objects according to the dependency between source and targets and foreign key nullability
+     * Skips deleting records present in $remainingEntities
+     *
+     * @param array $properties array of foreignKey properties
+     * @param EntityInterface $entity the entity which should have its associated entities unassigned
+     * @param Table $target The associated table
+     * @param array $remainingEntities Entities that should not be deleted
+     * @return void
+     */
+    protected function _unlinkAssociated(array $properties, EntityInterface $entity, Table $target, array $remainingEntities)
```

Comment:
Would it make sense to make this a little more `BelongsToMany`-ish, where linking/replacing/unlinking are public methods that can easily be used in app code?

https://github.com/cakephp/cakephp/blob/3.1.0/src/ORM/Association/BelongsToMany.php#L596
https://github.com/cakephp/cakephp/blob/3.1.0/src/ORM/Association/BelongsToMany.php#L637
https://github.com/cakephp/cakephp/blob/3.1.0/src/ORM/Association/BelongsToMany.php#L720

Being able to do stuff like


```php
$comments = [$comment1, $comment3];
$Articles->association('Comments')->replaceAssociated($article, $comments);
```

similar to `belongsToMany` associations might come handy, at least I could see _me_ using it :) It would also allow for easier testing that fits in `HasManyTest`.